src/inmanta/ast/entity.py

```diff
@@ -609,7 +610,8 @@ def pair_dataclass(self) -> None:
                     dc_fields.pop(rel_or_attr_name)
                     # Type correspondence
                     try:
-                        dsl_type = to_dsl_type(dc_types[rel_or_attr_name])
+                        plugin_line: Range = Range(self.location.file, self.location.lnr, 1, self.location.lnr + 1, 1)
```

Comment:
This name is confusing, because this is no plugin. That said, I think we can simply use `self.location` here. It will always be a `Range` in practice (locations are badly typed, little we can do about it, you can change the signature of `get_location()` here, which would help, but it would probably lead to a type error in its implementation, which is acceptable for now).

Reply:
Can't I just rename `plugin_line` to something like `location` (which would still be a range)? It feels wrong explicitly introducing typing errors

Reply:
I completely understand, but I think we have to consider the source of the type error. If it is due to a pre-existing issue, we're only introducing a new report for it, we're not really introducing a new issue. I think that context is really important when doing gradual typing, as (alas) we are doing. The type error does not make matters worse than they already are, it's just about accepting an unfortunate fact. In contrast, creating a new range here might in fact make matters worse, because we lose useful information: the range might point to a specific character, and we'd be widening that to a whole line.

Reply:
I think we specifically need a `Range` to create a `LocatableString` to pass onto `inmanta_type.resolve_type`, right?

Reply:
But we have one in `self.location`.